sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -0,0 +1,569 @@
+            self.gauge('memory.service.shared.used', shared_used, tags=tags, hostname=host)
+
+            shared_usable = shared_total - shared_used
+            self.gauge('memory.service.shared.usable', shared_usable, tags=tags, hostname=host)
+
+            if shared_total:
+                shared_utilized = shared_used / shared_total * 100
+            else:
+                shared_utilized = 0
+            self.gauge('memory.service.shared.utilized', shared_utilized, tags=tags, hostname=host)
+
+            # Compactors
+            compactors_total = memory['compactors_total']
+            self.gauge('memory.service.compactor.total', compactors_total, tags=tags, hostname=host)
+
+            compactors_usable = memory['compactors_usable']
+            self.gauge('memory.service.compactor.usable', compactors_usable, tags=tags, hostname=host)
+
+            compactors_used = compactors_total - compactors_usable
+            self.gauge('memory.service.compactor.used', compactors_used, tags=tags, hostname=host)
+
+            if compactors_total:
+                compactors_utilized = compactors_used / compactors_total * 100
+            else:
+                compactors_utilized = 0
+            self.gauge('memory.service.compactor.utilized', compactors_utilized, tags=tags, hostname=host)
+
+    def query_service_component_memory(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/20bed4f675191014a4cf8e62c28d16ae.html
+        for memory in self.iter_rows(queries.GlobalSystemServiceComponentMemory):
+            tags = [
+                'db:{}'.format(memory['db_name'] or 'none'),
+                'hana_port:{}'.format(memory['port']),
+                'component_name:{}'.format(memory['component']),
+            ]
+            tags.extend(self._tags)
+
+            self.gauge(
+                'memory.service.component.used',
+                memory['used'],
+                tags=tags,
+                hostname=self.get_hana_hostname(memory['host']),
+            )
+
+    def query_row_store_memory(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/20bb47a975191014b1e2f6bd0a685d7b.html
+        for memory in self.iter_rows(queries.GlobalSystemRowStoreMemory):
+            tags = [
+                'db:{}'.format(memory['db_name']),
+                'hana_port:{}'.format(memory['port']),
+                'resource_type:{}'.format(memory['category']),
+            ]
+            tags.extend(self._tags)
+
+            host = self.get_hana_hostname(memory['host'])
+
+            total = memory['total']
+            self.gauge('memory.row_store.total', total, tags=tags, hostname=host)
+
+            used = memory['used']
+            self.gauge('memory.row_store.used', used, tags=tags, hostname=host)
+
+            usable = memory['usable']
+            self.gauge('memory.row_store.usable', usable, tags=tags, hostname=host)
```

Comment:
if we're changing `usable` to `free` or `available` per alex's previous comment, this will need to updated too